src/components/SharedComponents/PermissionGateContainer.tsx

```diff
@@ -100,6 +104,9 @@ interface MultiResult {
   [permission: string]: PermissionStatus;
 }
 export function permissionResultFromMultiple( multiResults: MultiResult ) {
+  if ( !isTablet ) {
+    Orientation.lockToPortrait( );
+  }
```

Comment:
This is an odd place to put this. I can see why this addresses the issue (orientation gets locked before PermissionGate renders), but it also does this locking unnecessarily *every* time we process permission results.

Reply:
I understand that this might seem like an odd place to lock the orientation. I initially took reference from src/components/Camera/CameraWithDevice.tsx, where a similar approach was used to lock the orientation. However, I will dive deeper into the root cause of the issue and explore why its not working in the PermissionGateContainer